include/seqan3/container/detail/random_access_iterator.hpp

```diff
@@ -0,0 +1,235 @@
+    bool operator>(random_access_iterator const & rhs) const
+    {
+        return this->pos > rhs.pos;
+    }
+
+    //! \brief Iterator comparison refers to their positions
+    bool operator<=(random_access_iterator const & rhs) const
+    {
+        return this->pos <= rhs.pos;
+    }
+
+    //! \brief Iterator comparison refers to their positions
+    bool operator>=(random_access_iterator const & rhs) const
+    {
+        return this->pos >= rhs.pos;
+    }
+
+    //! \brief Pre-increment, return updated iterator
+    random_access_iterator & operator++()
+    {
+        ++pos;
+        return (*this);
+    }
+
+    //! \brief Post-increment, return previous iterator state
+    random_access_iterator operator++(int)
+    {
+        random_access_iterator cpy{*this};
+        ++pos;
+        return cpy;
+    }
+
+    //! \brief Pre-decrement, return updated iterator
+    random_access_iterator& operator--()
+    {
+        --pos;
+        return *this;
+    }
+
+    //! \brief Post-decrement, return previous iterator state
+    random_access_iterator operator--(int)
+    {
+        random_access_iterator cpy{*this};
+        --pos;
+        return cpy;
+    }
+
+    //! \brief Forward this iterator
+    random_access_iterator& operator+=(difference_type skip)
+    {
+        pos += skip;
+        return *this;
+    }
+
+    //! \brief Forward copy of this iterator
+    random_access_iterator operator+(difference_type skip) const
+    {
+        return random_access_iterator{*this->host, static_cast<position_type>(pos + skip)};
+    }
+
+    //! \brief Non-member operator+ delegates to non-friend operator+
+    friend random_access_iterator operator+(difference_type skip , const random_access_iterator& it)
+    {
+        return it + skip;
+    }
+
+    //! \brief Decrement iterator by skip
+    random_access_iterator& operator-=(difference_type skip)
+    {
+        pos -= skip;
+        return *this;
+    }
+
+    //! \brief Return decremented copy of this iterator
+    random_access_iterator operator-(difference_type skip) const
+    {
+        return random_access_iterator{*this->host, static_cast<position_type>(this->pos - skip)};
+    }
+
+    //! \brief Non-member operator- delegates to non-friend operator-
+    friend random_access_iterator operator-(difference_type skip , const random_access_iterator& it)
+    {
+        return it - skip;
+    }
+
+    //! \brief Return offset between this and remote iterator's position
+    difference_type operator-(random_access_iterator lhs) const
+    {
+        return static_cast<difference_type>(this->pos - lhs.pos);
+    }
+
+    //! \brief Return pointer to this iterator
+    pointer operator->() const
+    {
+        return &this->host[pos]; //&(*this);
+    }
+
+    //! \brief Return underlying container value currently pointed at
+    reference operator[](position_type const n) const
+    {
+        return (reference)(*host)[pos + n];
+    }
+};
```

Comment:
Since there are a lot of member functions, it would help the documentation, if they were grouped, see for example:
https://github.com/seqan/seqan3/blob/master/include/seqan3/alphabet/nucleotide/dna4.hpp
(the `\{` opens a group and `\}` closes the group).
I would propose a constructor/deconstructor group, dereference-group (*, -> and []), and a arithmetic operations group (++, --, -, +, +=, -=)
What do you think?